### PR TITLE
Eradicate sphinx warnings

### PR DESCRIPTION
**Pull request by chrishalcrow**
Screenshot 2024-07-09 at 14 20 10
->
Screenshot 2024-07-12 at 08 45 57

Update docs to deal with all sphinx warnings, except 1.

Mostly this is formatting and updates to to `:ref:`s, which control the links across the documentation. In #3181, it was agreed to use  `autosectionlabel_prefix_document`, which means the user needs to specify the file location when referring to a Section in another page of the documentation.

Also deleted the `quality_metrics` References page in favour of linking to the new References page.

The final warning is to do with the fact that sphinx-gallery cannot be pickled for caching. See https://github.com/sphinx-doc/sphinx/issues/12300 . We could turn off this warning using `suppress_warnings = ["config.cache"]`, but I'm not sure what else we'd be suppressing since I am clueless about our documentation caching.

Would appreciate if others tried running this locally and seeing how many warning they get (I had 1 on Mac, then 9 on Linux. Now 1 on each!). @zm711 @JoeZiminski @h-mayorquin Thanks!

### CODE REVIEW

File: doc/how_to/viewers.rst [hunk 1]
```diff
@@ -11,7 +11,7 @@ There are several ways to plot signals (raw, preprocessed) and spikes.
 4. You can use the :code:`phy` software
 
 
-spikeinterface.widgets
+spikeinterface-widgets
```
Comment (chrishalcrow) [inline doc/how_to/viewers.rst:14]:
```suggestion
spikeinterface.widgets
```

File: doc/modules/core.rst [hunk 2]
```diff
@@ -22,6 +22,7 @@ All classes support:
   * multiple segments, where each segment is a contiguous piece of data (recording, sorting, events).
 
 .. _core-recording:
```
Comment (chrishalcrow) [inline doc/modules/core.rst:24]:
```suggestion
```

**Comment by JoeZiminski** [comment]
I get these on Windows:
image


Also, do you experiecne the same problem of `..\examples\tutorials\core\plot_4_sorting_analyzer.py` always failing to build on Windows?

**Comment by alejoe91** [comment]
Thansk @JoeZiminski 

Honestly, I think it's ok to eradicate warnings only on Linux, since that willl be the OS used for building and deploying the docs.

**Comment by chrishalcrow** [comment]
> I get these on Windows: image


I now also get these - thought I was building everything from scratch but wasn't quite. Thanks!

**Comment by JoeZiminski** [comment]
I think for development @alejoe91 its useful, as if the default is to ignore warnings when developing the docs on Windows / macOS then errors introduced during development may not be caught (and then not seen during the automatic building on CI).

**Comment by chrishalcrow** [comment]
Could you have another go @JoeZiminski ? Thanks!

**Comment by JoeZiminski** [comment]
Hey @chrishalcrow I still get them strangely, let me know if you can't reproduce I'll look into it ASAP

image

I made a fresh env, and made sure `_build` was deleted before running.

<details>
<summary>my env<summary>
```
Package                       Version     Editable project location
----------------------------- ----------- -----------------------------------------------
alabaster                     0.7.16
annexremote                   1.6.5
asciitree                     0.3.3
asttokens                     2.4.1
Babel                         2.15.0
backports.tarfile             1.2.0
boto3                         1.34.143
botocore                      1.34.143
certifi                       2024.7.4
chardet                       5.2.0
charset-normalizer            3.3.2
click                         8.1.7
colorama                      0.4.6
contourpy                     1.2.1
cycler                        0.12.1
Cython                        0.29.37
datalad                       1.1.1
decorator                     5.1.1
distro                        1.9.0
docutils                      0.20.1
elephant                      1.1.0
exceptiongroup                1.2.1
executing                     2.0.1
fasteners                     0.19
fonttools                     4.53.1
h5py                          3.11.0
hdbscan                       0.8.37
humanize                      4.10.0
idna                          3.7
imagesize                     1.4.1
importlib_metadata            8.0.0
ipython                       8.26.0
iso8601                       2.1.0
jaraco.classes                3.4.0
jaraco.context                5.3.0
jaraco.functools              4.0.1
jedi                          0.19.1
Jinja2                        3.1.4
jmespath                      1.0.1
joblib                        1.4.2
keyring                       25.2.1
keyrings.alt                  5.0.1
kiwisolver                    1.4.5
lazy-ops                      0.2.0
llvmlite                      0.43.0
looseversion                  1.3.0
MarkupSafe                    2.1.5
matplotlib                    3.9.1
matplotlib-inline             0.1.7
MEArec                        1.9.1
MEAutility                    1.5.1
more-itertools                10.3.0
msgpack                       1.0.8
neo                           0.13.1
networkx                      3.3
numba                         0.60.0
numcodecs                     0.12.1
numpy                         1.26.4
numpydoc                      1.7.0
packaging                     24.1
pandas                        2.2.2
parso                         0.8.4
patool                        2.3.0
pillow                        10.4.0
pip                           24.0
platformdirs                  4.2.2
pooch                         1.8.2
probeinterface                0.2.21
prompt_toolkit                3.0.47
pure-eval                     0.2.2
Pygments                      2.18.0
pyparsing                     3.1.2
python-dateutil               2.9.0.post0
python-gitlab                 4.7.0
pytz                          2024.1
pywin32-ctypes                0.2.2
PyYAML                        6.0.1
quantities                    0.15.0
requests                      2.32.3
requests-toolbelt             1.0.0
s3transfer                    0.10.2
scikit-learn                  1.5.1
scipy                         1.14.0
setuptools                    69.5.1
six                           1.16.0
snowballstemmer               2.2.0
Sphinx                        7.3.7
sphinx_design                 0.6.0
sphinx-gallery                0.16.0
sphinx-rtd-theme              2.0.0
sphinxcontrib-applehelp       1.0.8
sphinxcontrib-devhelp         1.0.6
sphinxcontrib-htmlhelp        2.0.5
sphinxcontrib-jquery          4.1
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.7
sphinxcontrib-serializinghtml 1.1.10
spikeinterface                0.101.0rc0  C:\fMRIData\git-repo\forks\chris\spikeinterface
stack-data                    0.6.3
tabulate                      0.9.0
threadpoolctl                 3.5.0
tomli                         2.0.1
tqdm                          4.66.4
traitlets                     5.14.3
typing_extensions             4.12.2
tzdata                        2024.1
urllib3                       2.2.2
wcwidth                       0.2.13
wheel                         0.43.0
xarray                        2024.6.0
zarr                          2.17.2
zipp                          3.19.2
```

</details>

**Comment by chrishalcrow** [comment]
> I made a fresh env, am definately on the most recent commit, and made sure `_build` was deleted before running.

Just to 100% confirm: can you go into `doc/tutorials/comparison` and delete everything starting with `generated_` (this used to be automatically generated by sphinx, using the stuff in `examples/tutorials/comparison` and might not be deleted by any sphinx cleaning stuff). Then run again. Thanks!

**Comment by samuelgarcia** [comment]
merci Chris!

**Review by samuelgarcia — APPROVED** [review]

**Pull request merged**